### ISSUE TITLE
"useBuiltIns": "usage" does not add es6.set/es6.map polyfills

### ISSUE THREAD

**Issue by Daniellt1** [babel/babel#7335]
Set polyfill should be added in order to run React 16 application in Safari on iOS 7.

### Babel/Babylon Configuration (.babelrc, package.json, cli command)
.babelrc:
```js
{
  "presets": [
    [
      "@babel/preset-env",
      {
        "useBuiltIns": "usage",
        "modules": false,
        "loose": false,
        "debug": true
      }
    ],
    "@babel/preset-react"
  ],
  "plugins": [
    "react-hot-loader/babel",
    "dynamic-import-webpack",
    [
      "@babel/transform-runtime",
      {
        "helpers": false,
        "polyfill": false,
        "regenerator": true
      }
    ],
    [
      "transform-imports",
      {
        "validator": {
          "transform": "validator/lib/${member}",
          "preventFullImport": true
        }
      }
    ],
    [
      "direct-import",
      {
        "name": "redux-form/immutable",
        "indexFile": "redux-form/es/immutable"
      }
    ],
    ["transform-object-rest-spread", { "useBuiltIns": true }],
    "transform-class-properties",
    "@babel/transform-react-inline-elements",
    "transform-react-constant-elements"
  ],
  "comments": true
}
```

package.json browserslist:
```js
"browserslist": [
    "> 1%",
    "IE 11",
    "iOS >= 7"
  ]
```

### Expected Behavior
Set and Map polyfills which are required by React 16 should be added.

### Current Behavior
They are not added even with the explicit include:
```js
"include": ["es6.set", "es6.map"]
```

**Comment by existentialism** [comment]
@Daniellt1 can you show us the code that uses Map/Set or create a small repo that reproduces the issue?

It should work (though it's a trivial example):
https://babeljs.io/repl/#?babili=false&browsers=%3E%201%25%2C%20IE%2011%2C%20iOS%20%3E%3D%207&build=&builtIns=usage&code_lz=MYewdgzgLgBAZiEMC8MwFMDuMCyBDABwAoBKAbgChRJYAjPAJxTSxgGV0pTKg&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&lineWrap=false&presets=env&prettier=false&targets=&version=7.0.0-beta.39&envVersion=7.0.0-beta.39

**Comment by Daniellt1** [comment]
@existentialism,
Yeah, I see. You are absolutely correct. Do you have an idea why explicit "include" does not work then? Or I misunderstood the way it should work?

**Comment by existentialism** [comment]
@Daniellt1 the behavior of `include` and `exclude` were probably just not considered when we first landed `usage` (we haven't really made many updates to it since we first landed it)... probably because it might not know where to put them (i.e. entry knows because it found a `babel-polyfill` import/require).

**Comment by Daniellt1** [comment]
@existentialism,

Fair enough. I will add Set and Map polyfills to my entry points as a temporary fix. Do you consider to investigate it in the future?

**Comment by kaisermann** [comment]
I'm also facing the same dilemma while using `useBuiltIns: 'usage'`. For my project's code, it works awesomely, but I still need to __always__ include some polyfills for my frameworks.

**Comment by hzoo** [comment]
Thanks issue with usage + necessary polyfills is just that we don't have a place to put them in the code in babel itself unless this was something we did in another lib like babel-loader. We can't just put a polyfill whereever we want, and Babel operates per file. Usage works because it imports the polyfill at the top so it happens before the code is run. Babel doesn't know the entry point in the tool itself so there isn't really a solution atm. Otherwise you'd be running usage on node modules and we have to think about that more.

**Comment by Daniellt1** [comment]
@hzoo, I think it makes total sense and the issue can be closed. The only thing I want to mention is that `include` options have been working for me at the moment of this ticket creation. Not sure if that is working now. @kaisermann, how do you include the polyfills? Does `include` work for you?

**Comment by hzoo** [comment]
Yeah maybe the answer is that just need to error with include/exclude in usage mode since it has to be in an entry file for it to work at all

**Comment by hzoo** [comment]
Yeah I guess `exclude` would work (wasn't thinking about that I guess)

**Comment by nicolo-ribaudo** [comment]
Fixed in v7.4.2: https://babeljs.io/repl/version/7.4.2#?babili=false&browsers=%3E%201%25%2C%20IE%2011%2C%20IOS%20%3E%3D%207&build=&builtIns=usage&spec=false&loose=false&code_lz=HYUw7gBAsghgDgCgJQG4g&debug=true&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env&prettier=false&targets=&version=7.4.2
